godot-core/src/engine/script_instance.rs

```diff
@@ -382,10 +388,10 @@ mod script_instance_info {
     /// # Safety
     ///
     /// - The returned `*const T` is guaranteed to point to a list that has an equal length and capacity.
-    fn transfer_ptr_list_to_godot<T>(ptr_list: Vec<T>, list_length: &mut u32) -> *const T {
+    fn transfer_ptr_list_to_godot<T>(ptr_list: Box<[T]>, list_length: &mut u32) -> *const T {
         *list_length = ptr_list.len() as u32;
 
-        let ptr = Box::into_raw(ptr_list.into_boxed_slice());
+        let ptr = Box::into_raw(ptr_list);
 
         // SAFETY: `ptr` was just created in the line above and should be safe to dereference.
         unsafe { (*ptr).as_ptr() }
```

Comment:
I've actually just noticed that this introduces a subtle kind of UB:

Here you call `(*ptr).as_ptr()`. `as_ptr()` takes `&self` as an argument. This means that the returned pointer is derived from an immutable reference. Thus this pointer and any derived pointers cannot mutably access this box-slice.

However later you end up using this pointer to call `Vec::from_raw_parts`. This requires unique access to the boxed slice. But this pointer does not have permission to uniquely access that slice. Thus this is UB.

This can be fixed by changing this call to `(*ptr).as_mut_ptr()`. Ideally you should change the return type here to `*mut T` to reflect this change, as well as changing the corresponding argument `transfer_ptr_list_from_godot` type signature to reflect that as well. But that isn't safety critical.

Reply:
Fixed this.